### ISSUE TITLE
Convert scripts to `bash`

### ISSUE THREAD

**Issue by KateFriedman-NOAA** [NOAA-EMC/prepobs#36]
For better supportability on various platforms and consistency, scripts should be converted from `ksh`/`sh` to `bash`. Some are already `bash` but not all.

FYI, there was an issue with one of the `ksh` scripts in testing on Gaea. @DavidBurrows-NCO can provide a sample log with the silent failure that occurs. Tagging @ilianagenkova for her awareness.

**Comment by DavidBurrows-NCO** [comment]
Thanks @KateFriedman-NOAA... @ilianagenkova Kate suggested Hera would be a good platform to place the log on. I've put a gdasprep log at `/scratch1/NCEPDEV/nems/David.Burrows/for_iliana`. Permissions should be good for you to access.  The specific failure is `/gpfs/f5/ufs-ard/world-shared/global/glopara/data/git/prepobs/v1.0.2/install/ush/prepobs_makeprepbufr.sh[1620]: syntax error: unexpected operator/operand '=~'`. Thanks

**Comment by ilianagenkova** [comment]
Thanks for opening this issue!

@KateFriedman-NOAA , do you think (off the top of your head) this is similar problem to: 
[https://github.com/NOAA-EMC/global-workflow/issues/691]

**Comment by KateFriedman-NOAA** [comment]
@ilianagenkova Yes and no. That global workflow issue notes that error codes are not being caught on the way out so _any_ error in the prep job won't be caught. So yes, this is likely a silent failure because of that known issue but the error happening in this case is related to `ksh`.
